### PR TITLE
Update React Redux recipe to use custom render with wrapper

### PR DESCRIPTION
**Pull request by MatanBobi**
Updated the React Redux recipe to use a custom `render` with the `wrapper` option.
Also refactored the class component to a functional component.
This PR is following the talk with @kentcdodds in [this](https://github.com/reduxjs/redux/pull/3708) PR.

### CODE REVIEW

File: docs/example-react-redux.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Counter extends React.Component {
         <h2>Counter</h2>
         <div>
           <button onClick={this.decrement}>-</button>
-          <span data-testid="count-value">{this.props.count}</span>
+          <h1>{this.props.count}</h1>
```
Comment (alexkrolick) [inline docs/example-react-redux.md:26]: h1 after h2? Let's leave this alone.

Comment (MatanBobi) [reply docs/example-react-redux.md:26]: My bad, missed it sorry.. Will fix.

Comment (MatanBobi) [reply docs/example-react-redux.md]: @alexkrolick Fixed it, thanks..
Also refactored the class component to a functional one in the current time spirit :)

File: docs/example-react-redux.md [hunk 2]
```diff
@@ -58,58 +56,68 @@ export function reducer(state = initialState, action) {
...
+```
+
+```jsx
+// counter.test.js
+import React from 'react'
+import { createStore } from 'redux'
+import { Provider } from 'react-redux'
+import { fireEvent, screen } from '@testing-library/react'
+// We're using our own custom render function and not RTL's render
+import { render } from './test-utils.js
```
Comment (kentcdodds) [inline docs/example-react-redux.md:93]: In accordance with the setup suggestions, this should be:

```suggestion
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { render, fireEvent, screen } from './test-utils.js
```

Comment (MatanBobi) [reply docs/example-react-redux.md]: Updated it and also added the export to the `test-utils.js` file..
Thanks!

**Comment by kentcdodds** [comment]
@all-contributors please add @MatanBobi for docs

**Pull request merged**